src/mds/MDCache.cc

```diff
@@ -12084,6 +12084,23 @@ int MDCache::dump_cache(std::string_view fn, Formatter *f)
   int r = 0;
   int fd = -1;
 
+  // dumping large caches to formatter may cause mds to hang or worse
+  // get killed -- so, disallow the dump if the cache size exceeds the
+  // configured threshold, which is 1G (note that this can be jacked
+  // up by the admin... and is nothing but foot shooting, but the option
+  // itself is for devs and hence dangerous to tune). however, caches
+  // can still be dumped to a file, so, pretty print that in a message
+  // as a hint.
+  // TODO: remove this when fixed.
+  if (f && cache_size() >= g_conf().get_val<Option::size_t>("mds_dump_cache_threshold")) {
```

Comment:
Have you experimented with dumping to a file? Have you found any issues with 32GB+ size caches?

Reply:
yes, I did, but not with 32G+ cache size, but with a large enough cache that causes an issue when dumped to formatter.

Reply:
I'm concerned the issue may also exist when dumping to a file. Can you do some benchmarks to project at what size cache dumping to a file would be "slow" (5+ seconds)?

I'd also prefer the option name be more specific about what the limit is applied to. Let's append "_formatter"? I'd also suggest adding a separate limit for dumping to the file and just leave it unlimited if we can't presently find a good default.

Reply:
OK -- I assumed the issue was just somewhere in formatter. I'll perform the relevant tests for dumping to file to see what happens.